Scripts/Runtime/Entities/TaskDriver/TaskSet/Job/JobType/TaskJobForDeferExtension.cs

```diff
@@ -0,0 +1,135 @@
+            where TJob : struct, ITaskJobForDefer<TInstance>
+            where TInstance : unmanaged, IEntityProxyInstance
+        {
+            private const int UNSET_NATIVE_THREAD_INDEX = -1;
+
+            internal TJob JobData;
+            internal DataStreamActiveReader<TInstance> Reader;
+            [NativeSetThreadIndex] internal readonly int NativeThreadIndex;
+
+            public WrapperJobStruct(ref TJob jobData,
+                                    ref DataStreamScheduleInfo<TInstance> scheduleInfo)
+            {
+                JobData = jobData;
+                Reader = scheduleInfo.Reader;
+                NativeThreadIndex = UNSET_NATIVE_THREAD_INDEX;
+            }
+        }
+
+        //*************************************************************************************************************
+        // PRODUCER
+        //*************************************************************************************************************
+        private struct WrapperJobProducer<TJob, TInstance>
+            where TJob : struct, ITaskJobForDefer<TInstance>
+            where TInstance : unmanaged, IEntityProxyInstance
+        {
+            // ReSharper disable once StaticMemberInGenericType
+            internal static readonly IntPtr JOB_REFLECTION_DATA = JobsUtility.CreateJobReflectionData(typeof(WrapperJobStruct<TJob, TInstance>),
+                                                                                                      typeof(TJob),
+                                                                                                      (ExecuteJobFunction)Execute);
+
+
+            private delegate void ExecuteJobFunction(ref WrapperJobStruct<TJob, TInstance> jobData,
+                                                     IntPtr additionalPtr,
+                                                     IntPtr bufferRangePatchData,
+                                                     ref JobRanges ranges,
+                                                     int jobIndex);
+
+
+            [SuppressMessage("ReSharper", "MemberCanBePrivate.Global", Justification = "Required by Burst.")]
+            public static unsafe void Execute(ref WrapperJobStruct<TJob, TInstance> wrapperData,
+                                              IntPtr additionalPtr,
+                                              IntPtr bufferRangePatchData,
+                                              ref JobRanges ranges,
+                                              int jobIndex)
+            {
+                ref TJob jobData = ref wrapperData.JobData;
+                ref DataStreamActiveReader<TInstance> reader = ref wrapperData.Reader;
+                
+                jobData.InitForThread(wrapperData.NativeThreadIndex);
+
+                while (true)
+                {
+                    if (!JobsUtility.GetWorkStealingRange(ref ranges, jobIndex, out int beginIndex, out int endIndex))
```

Comment:
Why not put this as the while conditional?

Reply:
I don't know, this was just how Unity had set up their job types and I followed suit when first making this. 

I just changed them in all our jobs and they all still work, so yeah, good catch.